lib/material/internal/filebrowser/material_filebrowser_internal.flow

```diff
@@ -77,43 +77,87 @@ getLibraryUI(state : FilebrowserState<Pair<string, bool>, ?>, ext : [string], fi
 				contains(dirs, ".."),
 				previewsB
 			));
-			checkWhatIsChanged(currentPath, dirs, files, changesCallback)
+			checkWhatIsChanged(currentPath, dirs, files, state.engine, changesCallback)
 		},
 		onError
 	);
 
 	MMutable(viewB);
 }
 
+fileDatas(
+	fileName : string,
+	fileVersion : int
+);
+
 FSPathContent(
 	basePath : string,
 	dirs : [string],
-	files : [string]
+	files : [fileDatas]
 );
 
 lastCheckResultR : ref Maybe<FSPathContent> = ref None();
 
-checkWhatIsChanged(basePath : string, dirs : [string], files : [string], callback : ([string], [string]) -> void) -> void {
-	maybeApply(
+checkWhatIsChanged(basePath : string, dirs : [string], files : [string], engine : FilebrowserEngine<?, ??>, callback : ([string], [string], [string]) -> void) -> void {
+	updateLastCheckResult = \files3 -> {
+	eitherFn(
 		^lastCheckResultR,
 		\lcr -> {
 			if (basePath == lcr.basePath) {
+				files2 = map(lcr.files, \file -> file.fileName);
+
 				added = concat(
 					map(subtractA(dirs, lcr.dirs), \p -> basePath + p + "/"),
-					map(subtractA(files, lcr.files), \p -> basePath + p)
+					map(subtractA(files, files2), \p -> basePath + p)
 				);
+
 				removed = concat(
 					map(subtractA(lcr.dirs, dirs), \p -> basePath + p + "/"),
-					map(subtractA(lcr.files, files), \p -> basePath + p)
+					map(subtractA(files2, files), \p -> basePath + p)
 				);
 
-				if (added != [] || removed != [])
-					callback(added, removed);
-			}
-		}
-	);
+				updated = fold(
+					files3,
+					[],
+					\acc, f ->
+						eitherMap(
+							find(lcr.files, \f2 -> f2.fileName == f.fileName),
+							\f2 -> if(f2.fileVersion == f.fileVersion || f2.fileVersion == 0) acc else arrayPush(acc, f2),
+							acc
+						)
+				);
 
-	lastCheckResultR := Some(FSPathContent(basePath, dirs, files));
+				updatedNames = map(updated, \file -> file.fileName);
+				lastCheckResultR := Some(FSPathContent(basePath, dirs, files3));
+				if (added != [] || removed != []|| updatedNames != []) {
+					callback(added, removed, updatedNames);
+				}
+		}},
+		\-> lastCheckResultR := Some(FSPathContent(basePath, dirs, files3))
+	)}
+	mapAsync(
+		files,
+		\f, ff, rj -> engine.getExtraInfo(
+			basePath + f,
+			"version",
+			\m -> eitherFn(m, \m2 ->{ff(fileDatas(f, getJsonIntValue(m2, 0)))}, \-> rj("can't get version")),
+			rj
+		),
+
+
+		\files3 -> {
+			updateLastCheckResult(files3);
+		},
```

Comment:
\files3 -> {
			updateLastCheckResult(files3);
		},

is just updateLastCheckResult